File: Models/Group.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WeWakeAPI.Models
{
    public class Group
    {
        //public Group() {
        //    Members = new HashSet<User>();
        //}

        [Key]
        public Guid GroupId { get; set; }
        public string GroupName { get; set; }
        public Guid AdminId { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool CanMemberCreateAlarm { get; set; } = false;

        public Dictionary<String, object> AppendKey(string key, object obj)
        {
            var spreadObj = new Dictionary<string, object>();
            spreadObj.Add("groupId", GroupId);
            spreadObj.Add("groupName", GroupName);
            spreadObj.Add("adminId", AdminId);
            spreadObj.Add("createdAt", CreatedAt);
            spreadObj.Add("canMemberCreateAlarm", CanMemberCreateAlarm);
            spreadObj.Add(key, obj);

            return spreadObj;
        }

        override
        public string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }

    }

}

File: lib/models/Group.dart
class Group {
  String? GroupId;
  String? GroupName;
  String? AdminId;
  DateTime CreatedAt = DateTime.now();
  bool? CanMemberCreateAlarm;
  int? MemberCount = 1;
  bool IsAdmin = false;

  Group.fromJson(Map<String, dynamic> json) {
    GroupId = json['groupId'];
    GroupName = json['groupName'];
    AdminId = json['adminId'];
    CreatedAt = DateTime.parse(json['createdAt']);
    CanMemberCreateAlarm = json['canMemberCreateAlarm'];
    MemberCount = json['memberCount'];
    IsAdmin = json['isAdmin'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['groupId'] = this.GroupId;
    data['groupName'] = this.GroupName;
    data['adminId'] = this.AdminId;
    data['createdAt'] = this.CreatedAt;
    data['canMemberCreateAlarm'] = this.CanMemberCreateAlarm;
    data['memberCount'] = this.MemberCount;
    return data;
  }

  static List<Group> fromListJson(List<dynamic> arr) {
    List<Group> groups = [];
    for (var json in arr) {
      Group group = Group.fromJson(json);
      groups.add(group);
    }
    return groups;
  }
}
